backend/de.metas.business/src/main/java-legacy/org/compiere/process/RequisitionPOCreate.java

```diff
@@ -543,9 +543,9 @@ private boolean isVendorForProduct(final int C_BPartner_ID, final MProduct produ
 
 	}
 
-	private boolean isPOPriceList(final int M_PriceList_ID)
+	private boolean isPOPriceList(final int M_PriceList_ID, final I_M_Requisition requisition)
 	{
-		return Services.get(IQueryBL.class).createQueryBuilder(I_M_PriceList.class, getProcessInfo())
+		return Services.get(IQueryBL.class).createQueryBuilder(I_M_PriceList.class, requisition)
 				.addEqualsFilter(I_M_PriceList.COLUMNNAME_M_PriceList_ID, M_PriceList_ID)
```

Comment:
Please take out `Services.get(IQueryBL.class) `to the top and adjust the entire class.
The, use the query builder without giving it any object as a parameter. Shall get the trx from the thread if you use that.
So you do not need to give a new parameter only for the transaction.

Reply:
modification done